include/image.h

```diff
@@ -55,6 +60,282 @@
 #endif
 
 
+
+
+#if defined(ARCH_ARM) && defined(WOLFBOOT_ARMORED)
+
+struct wolfBoot_image {
+    uint8_t *hdr;
+    uint8_t *trailer;
+    uint8_t *sha_hash;
+    uint8_t *fw_base;
+    uint32_t fw_size;
+    uint32_t part;
+    uint32_t hdr_ok;
+    uint32_t canary_FEED4567;
+    uint32_t signature_ok;
+    uint32_t canary_FEED6789;
+    uint32_t not_signature_ok;
+    uint32_t canary_FEED89AB;
+    uint32_t sha_ok;
+};
+
+static void __attribute__((noinline)) wolfBoot_image_confirm_signature_ok(struct wolfBoot_image *img)
```

Comment:
Suggest adding macro for noinline like we do in wolfSSL:

```
#ifdef noinline
    #define WB_NOINLINE noinline
#elif defined(_MSC_VER)
    #define WB_NOINLINE __declspec(noinline)
#elif defined(__ICCARM__) || defined(__IAR_SYSTEMS_ICC__)
    #define WB_NOINLINE _Pragma("inline = never")
#elif defined(__GNUC__) || defined(__KEIL__) || defined(__DCC__)
    #define WB_NOINLINE __attribute__((noinline))
#else
    #define WB_NOINLINE
#endif
```

Reply:
ARMORED=1 code is now arm-gcc only